pkg/service/service.go

```diff
@@ -645,13 +645,26 @@ func (s *Service) upsertService(params *lb.SVC) (bool, lb.ID, error) {
 // filterServiceBackends returns the list of backends based on given front end ports.
 // The returned map will have key as port name/number, and value as list of respective backends.
 func filterServiceBackends(svc *svcInfo, onlyPorts []string) map[string][]*lb.Backend {
-	if len(onlyPorts) == 0 {
-		return map[string][]*lb.Backend{
-			anyPort: svc.backends,
+	hasPreferredBE := false
+	for _, backend := range svc.backends {
+		if backend.Preferred {
+			hasPreferredBE = true
+			break
 		}
 	}
 
 	res := map[string][]*lb.Backend{}
+	if len(onlyPorts) == 0 {
+		for _, backend := range svc.backends {
+			if hasPreferredBE && backend.Preferred == lb.Preferred(true) {
+				res[anyPort] = append(res[anyPort], backend)
+				continue
+			}
+			res[anyPort] = append(res[anyPort], backend)
```

Comment:
For me this looks like we could just return
```
map[string][]*lb.Backend{
			anyPort: svc.backends,
}
```
here, as it was before, because for every loop iteration we will add a backend to `res[anyPort]` in any case:
```
if (X): add backend
else: add backend
```

Reply:
good catch, I don't remember what I thought yesterday